CMakeLists.txt

```diff
@@ -208,6 +197,14 @@ else()
   message(STATUS "SIMD support disabled")
 endif()
 
+# Compiler options for different platforms and compilers
+if(MSVC)
+    target_compile_options(ipc_toolkit PRIVATE /W4 /MP)
```

Comment:
I would rather put these into the `ipc::toolkit::warnings` target. See `cmake/ipc_toolkit/ipc_toolkit_warnings.cmake`. Line 153 currently removes all warning flags if `MSVC`, but we could add these flags there and check that they are valid.